services-camel/src/test/java/edu/unc/lib/dl/services/camel/destroyDerivatives/DestroyDerivativesRouterIT.java

```diff
@@ -0,0 +1,345 @@
+    }
+
+    @Test
+    public void destroyCollectionNoImageTest() throws Exception {
+        CollectionObject collectionWithImg = repoObjectFactory.createCollectionObject(null);
+        adminUnit.addMember(collectionWithImg);
+
+        treeIndexer.indexAll(baseAddress);
+
+        initMarkForDeletionJob(collectionWithImg.getPid());
+        deletionJob.run();
+        initializeDestroyJob(Collections.singletonList(collectionWithImg.getPid()));
+        destroyJob.run();
+
+        verify(destroySmallThumbnailProcessor, never()).process(any(Exchange.class));
+        verify(destroyLargeThumbnailProcessor, never()).process(any(Exchange.class));
+        verify(destroyCollectionSrcImgProcessor, never()).process(any(Exchange.class));
+        verify(destroyAccessCopyProcessor, never()).process(any(Exchange.class));
+        verify(destroyFulltextProcessor, never()).process(any(Exchange.class));
+    }
+
+    @Test
+    public void destroyTextTest() throws Exception {
+        WorkObject work = repoObjectFactory.createWorkObject(null);
+        String mimetype = "text/plain";
+        FileObject fileObj = addFileToWork(work, mimetype);
+        work.addMember(fileObj);
+
+        treeIndexer.indexAll(baseAddress);
+
+        initMarkForDeletionJob(work.getPid());
+        deletionJob.run();
+        initializeDestroyJob(Collections.singletonList(fileObj.getPid()));
+        destroyJob.run();
+
+        verify(destroySmallThumbnailProcessor, never()).process(any(Exchange.class));
+        verify(destroyLargeThumbnailProcessor, never()).process(any(Exchange.class));
+        verify(destroyCollectionSrcImgProcessor, never()).process(any(Exchange.class));
+        verify(destroyAccessCopyProcessor, never()).process(any(Exchange.class));
+        verify(destroyFulltextProcessor).process(any(Exchange.class));
+    }
+
+    @Test
+    public void invalidTypeTest() throws Exception {
+        WorkObject work = repoObjectFactory.createWorkObject(null);
+        FileObject fileObj = addFileToWork(work, "application/octet-stream");
+        work.addMember(fileObj);
+
+        treeIndexer.indexAll(baseAddress);
+
+        initMarkForDeletionJob(fileObj.getPid());
+        deletionJob.run();
+        initializeDestroyJob(Collections.singletonList(fileObj.getPid()));
+        destroyJob.run();
+
+        verify(destroySmallThumbnailProcessor, never()).process(any(Exchange.class));
+        verify(destroyLargeThumbnailProcessor, never()).process(any(Exchange.class));
+        verify(destroyCollectionSrcImgProcessor, never()).process(any(Exchange.class));
+        verify(destroyAccessCopyProcessor, never()).process(any(Exchange.class));
+        verify(destroyFulltextProcessor, never()).process(any(Exchange.class));
+    }
+
+    private FileObject addFileToWork(WorkObject work, String mimetype) throws Exception {
+        FolderObject folder = repoObjectFactory.createFolderObject(null);
```

Comment:
Its a bit surprising that `addFileToWork` creates a folder, is that necessary? Works can be the children of collections